### DISCUSSION TITLE
Can one simulate diploid individuals?

### DISCUSSION THREAD

**Discussion by shaharMazia** [tskit-dev/msprime#2022] [Q&A] [answered]
Hi, I tried to address this question in [Biostars](https://www.biostars.org/p/9516645/) and didn't get an answer.
You are welcome to read the question (I prefer not to copy it here, but if it will be suggest, I will).

The current idea I have to deal with it, is to just join randomly genomes from the output to create diploid individuals. OF course to join genomes from the same sub populations. 

I am afraid though, my entire simulation, which has recombination, is not simulated correctly then.
Did anyone experience with backward simulations for diploids? Is that even possible with coalescent theory?

Thanks,
Shahar

**Comment by benjeffery** [comment, marked as answer]
Hi @shaharMazia!

You're using the legacy `simulate` function. It is highly recommended to use the updated methods of `sim_ancestry` and `sim_mutations`.
There is a `ploidy` argument to [sim_ancestry()](https://tskit.dev/msprime/docs/stable/api.html#msprime.sim_ancestry). Have you seen the section on ploidy in the docs which explains the effect of this in msprime?
https://tskit.dev/msprime/docs/stable/ancestry.html#sec-ancestry-ploidy

**Reply by shaharMazia** [reply]
Hi @benjeffery thanks for the very rapid response!
I noticed the ploidy arg for sim_ancestry(), but I am using msprime.simulate().
I am not sure what is the different, I learned to use the lib through [this paper](https://link.springer.com/protocol/10.1007/978-1-0716-0199-0_9) that uses the simulate() method.
Can I use [sim_ancestry()](https://github.com/tskit-dev/msprime/blob/9843e210b8b798935409bf539351325955c30d0e/msprime/ancestry.py#L1028) instead of [simulate()](https://github.com/tskit-dev/msprime/blob/9843e210b8b798935409bf539351325955c30d0e/msprime/ancestry.py#L408) and the rest will be the same? Sounds like a great solution if thats all, but some guts tells me backward simulation with recombination suppose to be a lot harder.

The docs says "The number of monoploid genomes per sample individual
        (Default=2). See the :ref:`sec_ancestry_ploidy` section for usage examples."
        
 It sounds like it is not exactly like diploid, but having 2 monoploid genomes per individuals (a bit like the hack I thought to do after the simulation). I couldn't find sec_ancestry_ploidy section, can you help me with that?
 
 Thanks a lot, Shahar

**Reply by shaharMazia** [reply]
Oh I just saw you answered as well in the Biostar post. 
Thanks, I will read a bit, and answer you there if I have more questions.

**Reply by benjeffery** [reply]
Yeah, I answered in both places in case someone else sees it over at BioStars. `simulate` was deprecated and replaced with `sim_ancestry` and `sim_mutations` as these fix some deficiencies and simplify the API. A paper updating the msprime one you linked to is at https://www.biorxiv.org/content/10.1101/2021.08.31.457499v3. `msprime` does the right thing with respect to polidy - it's not a hack!

The ploidy section of the docs is at this link: https://tskit.dev/msprime/docs/stable/ancestry.html#sec-ancestry-ploidy. `msprime` does the
